packages/gatsby/src/utils/page-data-helpers.ts

```diff
@@ -29,6 +31,11 @@ export function constructPageDataString(
     "matchPath": "${matchPath}"`
   }
 
+  if (manifestId) {
+    body += `,
+    "manifestId": "${manifestId}"`
```

Comment:
Ah, we ended up manually JSON stringifying anyway! 😄

Reply:
Why not create a JS object and use `JSON.stringify` on the whole thing?

Reply:
Manually stringifying is much faster here :D @wardpeet made this change 4 or 5 months ago to speed things up